torch/csrc/jit/codegen/cuda/expr_evaluator.cpp

```diff
@@ -10,10 +11,41 @@ namespace jit {
 namespace fuser {
 
 void EvaluationContext::bind(const Val* value, Int::ScalarType concrete_value) {
-  TORCH_CHECK(value->isAnInt());
-  TORCH_CHECK(!value->as<Int>()->value().has_value());
-  TORCH_CHECK(fusion_->origin(value) == nullptr);
-  bindings_[value] = concrete_value;
+  TORCH_INTERNAL_ASSERT(
+      value->isAnInt(),
+      "Expressoin Evaluation does not support values other than integers at this time.");
+
+  if (value->isConstScalar()) {
+    auto const_value = value->as<Int>()->value().value();
+    TORCH_INTERNAL_ASSERT(
+        concrete_value == const_value,
+        "Tried to bind ",
+        concrete_value,
+        " to ",
+        value,
+        " however ",
+        value,
+        " is set to a constant ",
+        const_value);
+  }
+
+  TORCH_INTERNAL_ASSERT(
+      fusion_->origin(value) == nullptr,
+      "Tried to bind to a value that is computed in the fusion IR. ",
+      "Can only bind to symbolic values to the fusion that do not have an origin expr.");
+
+  if (bindings_.find(value) != bindings_.end()) {
```

Comment:
as we discussed, I think this check is use-case specific and belongs outside the evaluator